app/controllers/UserController.java

```diff
@@ -0,0 +1,127 @@
+package controllers;
+
+import forms.LoginForm;
+import forms.RegisterForm;
+import models.User;
+import play.data.Form;
+import play.data.FormFactory;
+import play.db.jpa.Transactional;
+import play.libs.Json;
+import play.mvc.Result;
+import services.UserService;
+
+import javax.inject.Inject;
+
+public class UserController extends BaseController {
+
+    private UserService userService;
+    private FormFactory formFactory;
+
+    /**
+     * Sets service which will then be used
+     * to communicate with the database.
+     * @param userService
+     */
+    @Inject
+    public void setUserService(UserService userService) {
+        this.userService = userService;
+    }
+
+    /**
+     * Sets form factory which will then be used
+     * to create needed forms for creating User objects
+     * out of the request data.
+     * @param formFactory
+     */
+    @Inject
+    public void setFormFactory(FormFactory formFactory) {
+        this.formFactory = formFactory;
+    }
+
+    /**
+     * Method is called when a POST request is sent to
+     * /api/v1/register, it is used to call a service to
+     * write a user in the database, and also give the
+     * registered user the current session.
+     * @return the result
+     */
+    @Transactional
+    public Result register() {
+        try {
+            Form<RegisterForm> registerForm = formFactory.form(RegisterForm.class);
+
+            User user = userService.register(registerForm.bindFromRequest().get());
+
+            setToken(user);
+
+            return ok(Json.toJson(user));
+        } catch (Exception e) {
+            return badRequest(e.getMessage());
+        }
+    }
+
+    /**
+     * Method is called when a POST request is sent to
+     * /api/v1/login, it is used to give the current
+     * session to the logged in user.
+     * @return
+     */
+
+    @Transactional(readOnly = true)
+    public Result login() {
+        try {
+            Form<LoginForm> loginForm = formFactory.form(LoginForm.class);
+
+            User user = userService.login(loginForm.bindFromRequest().get());
```

Comment:
Kad bind-aš formu trebao bi provjeriti da li ima grešaka, možeš dodati anotaciju na model koja kaže koji su field-ovi obavezni za formu, tako da imaš validaciju da je stvarno poslano sve što treba